Tools/scripts/freeze_modules.py

```diff
@@ -710,18 +715,19 @@ def regen_pcbuild(modules):
 #######################################
 # the script
 
-def main():
+def main(deepfreeze_only: bool):
     # Expand the raw specs, preserving order.
     modules = list(parse_frozen_specs())
 
     # Regen build-related files.
     regen_makefile(modules)
     regen_pcbuild(modules)
-    regen_frozen(modules)
+    regen_frozen(modules, deepfreeze_only)
 
 
 if __name__ == '__main__':
-    argv = sys.argv[1:]
-    if argv:
-        sys.exit(f'ERROR: got unexpected args {argv}')
-    main()
+    parser = argparse.ArgumentParser()
+    parser.add_argument("--deepfreeze-only", action="store_true",
+        help="Only use deepfrozen modules", default=True)
```

Comment:
Um, since the default is True and the action is store_true, how do you ever disable this flag? Is there some secret argparse feature I don't know?

Reply:
Fixed here https://github.com/python/cpython/pull/31131